sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py

```diff
@@ -134,16 +136,21 @@ def begin_recognize_receipts(self, receipt, **kwargs):
         if content_type is None:
             content_type = get_content_type(receipt)
 
-        return self._client.begin_analyze_receipt_async(  # type: ignore
-            file_stream=receipt,
-            content_type=content_type,
-            include_text_details=include_field_elements,
-            cls=kwargs.pop("cls", self._receipt_callback),
-            polling=LROBasePolling(timeout=polling_interval, **kwargs),
-            error_map=error_map,
-            continuation_token=continuation_token,
-            **kwargs
-        )
+        try:
+            return self._client.begin_analyze_receipt_async(  # type: ignore
+                file_stream=receipt,
+                content_type=content_type,
+                include_text_details=include_field_elements,
+                cls=kwargs.pop("cls", self._receipt_callback),
+                polling=LROBasePolling(timeout=polling_interval, **kwargs),
+                error_map=error_map,
+                continuation_token=continuation_token,
+                **kwargs
+            )
+        except HttpResponseError as e:
+            ret_val = cast(LROPoller, None)
+            process_form_exception(e)
+            return ret_val
```

Comment:
why not just do `return process_form_exception(e)`? we're already ignoring the return type because of the generated code mismatch so I'm not sure this is adding much for mypy.

Reply:
good idea - will change everywhere